### PR TITLE
 Junit Test loading lots of files when isInspectRule is active #4 

### PR DESCRIPTION
**Pull request by Hitogoroshi**
The test was created and no errors were generated.

### CODE REVIEW

File: src/test/java/fr/jmmc/oitools/IsInspectRuleTest.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    @Test
+    public void dumpFile() throws IOException, FitsException {
+
+        final OIFitsChecker checker = new OIFitsChecker();
+        List<String> failureMsgs = new ArrayList<String>();
+        String fileName = "";
+        try {
+            for (String f : getFitsFiles(new File(TEST_DIR_OIFITS))) {
+
+                OIFitsChecker.setInspectRules(true);
```
Comment (bourgesl) [inline src/test/java/fr/jmmc/oitools/IsInspectRuleTest.java]: OIFitsChecker.setInspectRules(true) is critical => must be called BEFORE anything else
It is used to initialize Rule.applyTo (set) => to be verified

File: src/test/java/fr/jmmc/oitools/IsInspectRuleTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            logger.log(Level.SEVERE, "exception:", e);
+            failureMsgs.add("File: " + fileName + "throws an exception : " + e.getMessage() + "\n");
+        } finally {
+            OIFitsChecker.setInspectRules(false);
+        }
+        if (!failureMsgs.isEmpty()) {
+            for (String failureMsg : failureMsgs) {
+                Assert.fail(failureMsg);
+            }
+        }
```
Comment (bourgesl) [inline src/test/java/fr/jmmc/oitools/IsInspectRuleTest.java]: TODO: Check that Rule.applyTo() are not empty to ensure inspectRules() worked properly

File: src/test/java/fr/jmmc/oitools/IsInspectRuleTest.java (hunk 1 above)
Comment (bourgesl) [reply src/test/java/fr/jmmc/oitools/IsInspectRuleTest.java]: Try at the beginning of the method before new OIFitsChecker() 
or in junit setupClass() if you prefer

**Comment by Hitogoroshi** [comment]
OIFitsChecker.setInspectRules(true) is before the new new OIFitsChecker().
And I have check Rule.applyTo() has been created.

I tried to recreate the error in parseColumn and the test did not work anymore !

**Comment by bourgesl** [comment]
Bravo : 1st_place_medal:

**Pull request merged**